plugin/src/main/java/hudson/plugins/swarm/PluginImpl.java

```diff
@@ -54,9 +57,28 @@ public void doCreateSlave(StaplerRequest req, StaplerResponse rsp, @QueryParamet
 				nodeProperties = Lists.newArrayList(new ToolLocationNodeProperty(parsedToolLocations));
             }
     		
-            // try to make the name unique. Swarm clients are often replicated VMs, and they may have the same name.
-            if (jenkins.getNode(name) != null) {
-                name = name + '-' + req.getRemoteAddr();
+            if (hash == null && jenkins.getNode(name) != null) {
```

Comment:
does this mean that no existing swarm clients will be able to reconnect when Jenkins restarts - or when a swarm node disconnects its node is removed (and hence there would be no previous swamr nodes on a Jenkins restart?)